### PR TITLE
Ui nav

### PR DESCRIPTION
**Pull request by err**

### CODE REVIEW

File: src/client/app/mutations.cljs [hunk 1]
```diff
@@ -22,3 +22,12 @@
...
                                   (dissoc :all-items))))))})
+
+
+(defmethod m/mutate 'app/choose-tab [{:as env
+                                      :keys [state ;; app-state atom
+                                             ref]} ;; ident of component on which mutation is run
+
+                                     dispatch-key ;; 'app/choose-tab
+                                     {:keys [tab] :as params}]
+  {:action (fn [] (swap! state assoc-in [:tabs 0] tab))})
```
Comment (err) [inline src/client/app/mutations.cljs:33]: the way this works is as follows:

the current tab is represented by the first element of the ident associated with the `:tabs` key in app-state.

Take the mutation `[(app/choose-tab {:tab :data-tab})]`
prior to this mutation, app state may look as follows:

``` clj
{:tabs [:todo-tab 1]
 :todo-tab {1 {:id 1 :lists [[lists/by-title "Initial List"]]}}}
 :data-tab {1 {:id 1 :data [[viz/by-id 1]]}}}
```

and in the mutation, we update the element at the `0`th index of the `:tabs` ident, 
so now app-state is:

``` clj
{:tabs [:data-tab 1]  ;; huzzah
 :todo-tab {1 {:id 1 :lists [[lists/by-title "Initial List"]]}}}
 :data-tab {1 {:id 1 :data [[viz/by-id 1]]}}}
```

~~and the~~

File: src/client/app/ui.cljs [hunk 2]
```diff
@@ -41,19 +41,126 @@
...
+  Object
+  (render [this]
+          (let [{:keys [lists]} (om/props this)]
+            (dom/div nil
+                     (dom/b nil (str "TodoTab lists: " (pr-str lists)))
+                     (map ui-list lists)))))
+(def ui-todo-tab (om/factory TodoTab {:keyfn :id}))
+
+
+(defui ^:once DataPoint
```
Comment (err) [inline src/client/app/ui.cljs:67]: i don't think it was necessary to model it to this fine granularity, but it was a good way to exercise some of the (de)composition.

a `DataTab` may contain many `DataViz`(ualizations), each of which may be comprised of multiple `DataSeries` which is a sequence of `DataPoint`s ... as generic as I care to take this for now. We're more likely to end up with `D3Component` which takes a blob of app-state and we treat like a black box.

Comment (err) [inline src/client/app/ui.cljs:67]: ofc. my comment refers to code that's out of frame, starting on line 67

**Comment by halcyon** [comment]
👍

**Pull request merged**